languages.toml

```diff
@@ -144,7 +144,8 @@ language-server = { command = "typescript-language-server", args = ["--stdio"] }
 indent = { tab-width = 2, unit = "  " }
 
 [[language]]
-name = "tsx"
+name = "typescriptreact"
```

Comment:
I don't quite like this since now our language names are tied to what the LSP expects (which might also be different per language server). Let's instead add a `language-id` key to https://github.com/helix-editor/helix/blob/01782ee16adcffd56af44890ad772477dbefe610/helix-core/src/syntax.rs#L93-98

```rust
pub language_id: Option<String>
```

Then the lsp code can use `config.language_server.language_id.unwrap_or(config.name)`

Reply:
Yeah, I thought so as well (I briefly mentioned it in the pull request description). I'll update the PR. 🙂

Reply:
Hmm, should it really be part of the LSP settings and not the language itself? It's not really a configuration of the LSP, but more specifically an identifier of the language itself. I think it would make more sense to put `language-id` under the `language` section.

Thoughts?

Reply:
It's a lsp-specific language identifier. There's no standard conventions here, so I'd prefer to keep it grouped under the language server. It's quite likely that there could be another Javascript language server that expects a different identifier

Reply:
Fixed in https://github.com/helix-editor/helix/pull/1466/commits/c42377cc1dc12f3f97f4233fa941da04996f1941.